core/src/com/unciv/logic/civilization/transients/CivInfoTransientCache.kt

```diff
@@ -215,23 +217,43 @@ class CivInfoTransientCache(val civInfo: Civilization) {
             civInfo.addNotification("We have discovered [${tile.naturalWonder}]!",
                 tile.position, NotificationCategory.General, "StatIcons/Happiness")
 
-            var goldGained = 0
+            var relevantStat: Stat? = null
+            var statGained = 0
+
             val discoveredNaturalWonders = civInfo.gameInfo.civilizations.filter { it != civInfo && it.isMajorCiv() }
                     .flatMap { it.naturalWonders }
-            if (tile.terrainHasUnique(UniqueType.GrantsGoldToFirstToDiscover)
+            if (tile.terrainHasUnique(UniqueType.GrantsStatToFirstToDiscover)
                     && !discoveredNaturalWonders.contains(tile.naturalWonder!!)) {
-                goldGained += 500
+
+                for (unique in tile.getTerrainMatchingUniques(UniqueType.GrantsStatToFirstToDiscover)) {
+                    relevantStat = Stat.safeValueOf(unique.params[1])
+                    val bonusAmount = unique.params[0].toInt()
+
+                    statGained += bonusAmount
+                }
             }
 
-            if (civInfo.hasUnique(UniqueType.GoldWhenDiscoveringNaturalWonder)) {
-                goldGained += if (discoveredNaturalWonders.contains(tile.naturalWonder!!)) 100 else 500
+            if (civInfo.hasUnique(UniqueType.StatBonusWhenDiscoveringNaturalWonder)) {
+                for (unique in civInfo.getMatchingUniques(UniqueType.StatBonusWhenDiscoveringNaturalWonder)) {
+
+                    if (relevantStat == null)
+                        relevantStat = Stat.safeValueOf(unique.params[1])
+
+                    val normalBonus = unique.params[0].toInt()
+                    val firstDiscoveredBonus = unique.params[2].toInt()
+
+                    statGained += if (discoveredNaturalWonders.contains(tile.naturalWonder!!))
+                        normalBonus
+                    else
+                        firstDiscoveredBonus
+                }
             }
 
-            if (goldGained > 0) {
```

Comment:
Do not replace existing unique implementation! This will break all current mods!
Instead add implementation for new unique alongside the old one